src/Orleans/Messaging/IncomingMessageBuffer.cs

```diff
@@ -43,6 +39,12 @@ public IncomingMessageBuffer(TraceLogger logger, bool supportForwarding = false,
             bodyLength = 0;
         }
 
+        public List<ArraySegment<byte>> ReceiveBuffer()
```

Comment:
Keeping the same name as the property for the method is a little misleading, as this is not receiving a buffer. Maybe `GetReceiveBuffer()` or `BuildReceiveBuffer()`